core/state/journal.go

```diff
@@ -234,3 +242,43 @@ func (ch addPreimageChange) revert(s *StateDB) {
 func (ch addPreimageChange) dirtied() *common.Address {
 	return nil
 }
+
+func (ch accessListAddAccountChange) revert(s *StateDB) {
+	/*
+		One important invariant here, is that whenever a (addr, slot) is added, if the
+		addr is not already present, the add causes two journal entries:
+		- one for the address,
+		- one for the (address,slot)
+		Therefore, when unrolling the change, we can always blindly delete the
+		(addr) at this point, since no storage adds can remain when come upon
+		a single (addr) change.
+	*/
+	delete(s.accessList.addresses, *ch.address)
+}
+
+func (ch accessListAddAccountChange) dirtied() *common.Address {
+	return nil
+}
+
+func (ch accessListAddSlotChange) revert(s *StateDB) {
+	idx, addrOk := s.accessList.addresses[*ch.address]
+	if !addrOk {
+		panic("reverting slot change, address not present in list")
+	}
+	if idx < 0 {
+		panic("reverting slot change, slot not present list")
+	}
```

Comment:
You don't really need these two extra panics (unless you want to be explicit about the message), and they are going to panic anyway on their own when accessing.

Reply:
The `!addrOk` might not panic though, since it will just leave the `idx` zero. And then things will be strange, because some other slotmap may be deleted. So it's better to leave that panic in, until we're certain that it indeed does not happen